src/components/Block.js

```diff
@@ -0,0 +1,24 @@
+import React, {Component} from "react";
+import Map from "./Map";
+import Header from "./Header";
+
+export default class Block extends Component {
+
+    render() {
+        console.log(this.props);
+        const {blocks} = this.props;
+
+        const {name} = this.props.match.params;
+        const blockTmp = blocks.filter( el => el.name.toLowerCase() === name.toLowerCase());
```

Comment:
use `find` array method (`blocks.find(...)`), it will find one element for given condition (predicate). Then you'd be able to remove `blockTmp[0]` :) 
docs 👉 https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

PS please don't name variables like `blockTmp`, same as `foo`, `bar` etc :) give them meaningfull names!

Reply:
why find  method is better than filter  ??

Reply:
`find` returns first element from array which satisfies conditions -> which is clearly what you're trying to do here.

Moreover -> you're using here array as data structure which "has to have at least one element", when in general empty array is totally valid array.
If we need to find element `find` is clear message - give me one, which will have value or be `undefined` if not found.

No performance gain (at least non while working on tables smaller than 10k (or even 100k) elements) here in `find` vs `filter`.

Reply:
Example:
![image](https://user-images.githubusercontent.com/6861120/76982816-b7c02a80-693c-11ea-86fe-6a5de283e824.png)